plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaSocketLanguageServerLauncher.java

```
@@ -0,0 +1,148 @@
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.reflect.Proxy;
+import java.net.Socket;
+import java.util.Arrays;
+import javax.annotation.PreDestroy;
+import org.eclipse.che.api.languageserver.exception.LanguageServerException;
+import org.eclipse.che.api.languageserver.registry.DocumentFilter;
+import org.eclipse.che.api.languageserver.registry.LanguageServerDescription;
+import org.eclipse.che.api.languageserver.remote.CustomSocketLanguageServerLauncher;
+import org.eclipse.che.api.languageserver.remote.SocketLanguageServerLauncher;
+import org.eclipse.che.api.languageserver.service.FileContentAccess;
+import org.eclipse.che.api.languageserver.util.DynamicWrapper;
+import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.lsp4j.services.LanguageServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This launcher allows jdt.ls to be wrapped by the jdt.ls extension when launched via socket
+ *
+ * @author Joshua Pinkney
+ */
+@Singleton
+public class JavaSocketLanguageServerLauncher extends SocketLanguageServerLauncher
+    implements CustomSocketLanguageServerLauncher {
+  private static final Logger LOG = LoggerFactory.getLogger(JavaLanguageServerLauncher.class);
+  private int port;
+  private String host;
+  private Socket socket;
+
+  private static final LanguageServerDescription DESCRIPTION = createServerDescription();
+
+  private ProcessorJsonRpcCommunication processorJsonRpcCommunication;
+
+  @Inject
+  public JavaSocketLanguageServerLauncher(
+      ProcessorJsonRpcCommunication processorJsonRpcCommunication) {
+    super(null, "", 0);
+    this.processorJsonRpcCommunication = processorJsonRpcCommunication;
+  }
+
+  @PreDestroy
+  public void closeSocket() {
+    try {
+      socket.close();
+    } catch (IOException e) {
+
```

Eating exceptions should only happen if you know exactly what is happening. In this case, you eat it because you say we don't care, but someone else might care, so please log it as a warning